### PR TITLE
Coupling fields: Support Piola mapping

### PR DESCRIPTION
**Pull request by akva2**
Running staggered Boussinesq simulations with Piola mapped velocities we need to apply the forward Piola mapping when evaluating the velocity field in the temperature advection equation.

I'm not too thrilled about putting the flag in the ASM but I cannot see any cleaner way. At least it's sort of kept private, but the field and ASM both know the secret.

### CODE REVIEW

**Comment by akva2** [comment]
jenkins build this with downstreams please

**Comment by akva2** [comment]
jenkins build this with downstreams please

**Review by kmokstad — COMMENTED** [review]
Yep, maybe not the optimal solution for the piola variable. Now it will work only if you invoke the field evaluation method after the integrate method for the same integrand, and no other no-piola integrand in-between. I guess that scenario is not that likely, so this can pass for now.

File: src/ASM/ASMs2Dmx.C [hunk 1]
```diff
@@ -1062,23 +1062,24 @@ bool ASMs2Dmx::evalSolutionPiola (Matrix& sField, const Vector& locSol,
...
     return false;
   }
+  bool withPressure = len == locSol.size();
 
   const Go::SplineSurface* geo = this->getBasis(ASM::GEOMETRY_BASIS);
 
   if (regular)
   {
-    for (size_t b = 0; b < m_basis.size(); ++b)
+    for (size_t b = 0; b < (withPressure ? 3 : 2); ++b)
```
[the remark this answers is not in the document]
Comment (akva2) [reply src/ASM/ASMs2Dmx.C]: did this except nBasis instead of nb to avoid shadowing the member array.

**Comment by akva2** [comment]
jenkins build this with downstreams please

**Comment by akva2** [comment]
jenkins build this with downstreams please

**Pull request merged**